src/main/java/racingcar/domain/RacingCar.java

```diff
@@ -2,38 +2,60 @@
 
 import java.util.Objects;
 
-public class RacingCar {
+public class RacingCar implements Comparable<RacingCar> {
 
+    private static final String racingCarNamePattern = "^[a-zA-Z0-9]{1,4}$";
+    private String carName;
     private int carPosition;
 
-    public RacingCar() {
+    public RacingCar(String carName) {
+        this.carName = carName;
     }
 
-    public RacingCar(int carPosition) {
+    public RacingCar(String carName, int carPosition) {
+        this.carName = carName;
         this.carPosition = carPosition;
     }
 
     public RacingCar run(MoveRule racingRule) {
-        if(racingRule.checkMovement()) {
+        if(racingRule.isAbleToMove()) {
             carPosition++;
         }
-        return new RacingCar(this.carPosition);
+        return new RacingCar(this.carName, this.carPosition);
+    }
+
+    public String getCarName() {
+        return carName;
     }
 
     public int getCarPosition() {
         return carPosition;
     }
 
+    public static boolean validateCarName(String racingCarName) {
+        return racingCarName.matches(racingCarNamePattern);
+    }
+
+    public boolean isSamePosition(RacingCar other) {
+        return other.getCarPosition() == this.carPosition;
```

Comment:
같은 클래스는 getter 없이 필드 접근이 가능합니다.

```suggestion
        return other.carPosition == this.carPosition;
```

Reply:
수정하겠습니다.